cypress/integration/save.cy.js

```diff
@@ -0,0 +1,188 @@
+import {
+    DIMENSION_ID_DATA,
+    isYearOverYear,
+    visTypeDisplayNames,
+} from '@dhis2/analytics'
+
+import { openDimension } from '../elements/DimensionsPanel'
+import {
+    selectDataElements,
+    clickDimensionModalUpdateButton,
+    selectRelativePeriods,
+} from '../elements/DimensionModal'
+import { changeVisType } from '../elements/VisualizationTypeSelector'
+import {
+    expectAOTitleToBeDirty,
+    expectAOTitleToBeUnsaved,
+    expectAOTitleToBeValue,
+    expectAOTitleToNotBeDirty,
+    expectVisualizationToBeVisible,
+} from '../elements/Chart'
+import { TEST_DATA_ELEMENTS } from '../utils/data'
+import {
+    //openRandomAOCreatedByOthers,
+    saveNewAO,
+    closeFileMenu,
+    saveAOAs,
+    saveExistingAO,
+    openAOByName,
+    deleteAO,
+    expectFileMenuButtonToBeEnabled,
+    expectFileMenuButtonToBeDisabled,
+    FILE_MENU_BUTTON_SAVEAS,
+    FILE_MENU_BUTTON_DELETE,
+    FILE_MENU_BUTTON_RENAME,
+    FILE_MENU_BUTTON_TRANSLATE,
+    FILE_MENU_BUTTON_SHARE,
+    FILE_MENU_BUTTON_GETLINK,
+} from '../elements/FileMenu'
+import { expectRouteToBeAOId, expectRouteToBeEmpty } from '../elements/Route'
+import { getRandomVisType } from '../utils/random'
+import {
+    clickMenuBarFileButton,
+    clickMenuBarUpdateButton,
+} from '../elements/MenuBar'
+import { selectYoyCategoryOption } from '../elements/Layout'
+import {
+    expectStartScreenToBeVisible,
+    goToStartPage,
+} from '../elements/StartScreen'
+
+const TEST_VIS_NAME = `TEST ${new Date().toLocaleString()}`
+const TEST_VIS_NAME_UPDATED = `${TEST_VIS_NAME} - updated`
+const TEST_VIS_DESCRIPTION = 'Generated by Cypress'
+const TEST_VIS_TYPE = getRandomVisType()
+const TEST_VIS_TYPE_NAME = visTypeDisplayNames[TEST_VIS_TYPE]
+
+// TODO: Add test to check that the description is saved and shown in the interpretations panel
+
+describe('saving an AO', () => {
+    describe('"save" and "save as" for a new AO', () => {
+        it('navigates to the start page', () => {
+            goToStartPage()
+        })
+        it(`changes vis type to ${TEST_VIS_TYPE_NAME}`, () => {
+            changeVisType(TEST_VIS_TYPE_NAME)
+        })
+        it('adds Data dimension items', () => {
+            openDimension(DIMENSION_ID_DATA)
+            selectDataElements(
+                TEST_DATA_ELEMENTS.slice(1, 2).map(item => item.name)
+            )
+            clickDimensionModalUpdateButton()
+        })
+        it('displays an unsaved visualization', () => {
+            expectVisualizationToBeVisible(TEST_VIS_TYPE)
+            expectAOTitleToBeUnsaved()
+            expectRouteToBeEmpty()
+        })
+        it('checks that Save as is disabled', () => {
+            clickMenuBarFileButton()
+            expectFileMenuButtonToBeDisabled(FILE_MENU_BUTTON_SAVEAS)
+            closeFileMenu()
+        })
+        it('saves new AO using "Save"', () => {
+            saveNewAO(TEST_VIS_NAME, TEST_VIS_DESCRIPTION)
```

Comment:
Is this AO ever deleted?

Reply:
Yeah its operating on the same AO the whole time. First it does a "Save" of the unsaved state, then a regular "Save" of the updates, then a "Save as" which renames it, then a "Delete". This breaks the test interdependency best practice but in turn also makes sure that tests are self-cleaning.